app/assets/javascripts/controllers/network_router/network_router_form_controller.js

```diff
@@ -41,6 +46,30 @@ ManageIQ.angular.app.controller('networkRouterFormController', ['$http', '$scope
     miqService.miqAjaxButton(url, $scope.networkRouterModel, { complete: false });
   };
 
+  $scope.addInterfaceClicked = function() {
+    miqService.sparkleOn();
+    var url = '/network_router/add_interface/' + networkRouterFormId + '?button=add';
+    miqService.miqAjaxButton(url, $scope.networkRouterModel, { complete: false });
+  };
+
+  $scope.removeInterfaceClicked = function() {
+    miqService.sparkleOn();
+    var url = '/network_router/remove_interface/' + networkRouterFormId + '?button=remove';
+    miqService.miqAjaxButton(url, $scope.networkRouterModel, { complete: false });
+  };
+
+  $scope.cancelAddInterfaceClicked = function() {
+    miqService.sparkleOn();
+    var url = '/network_router/add_interface/' + networkRouterFormId + '?button=cancel';
+    miqService.miqAjaxButton(url);
+  };
+
+  $scope.cancelRemoveInterfaceClicked = function() {
+    miqService.sparkleOn();
+    var url = '/network_router/remove_interface/' + networkRouterFormId + '?button=cancel';
+    miqService.miqAjaxButton(url);
+  };
+
```

Comment:
Where are the `cancelAddInterfaceClicked()` and `cancelRemoveInterfaceClicked()` routines called from?